### PR TITLE
Allow export default abstract class. Related to issue 3792.

### PR DESCRIPTION
**Pull request by erikmcc**
Fixes (part of) #3792

### CODE REVIEW

File: tests/baselines/reference/classAbstractManyKeywords.errors.txt [hunk 1]
```diff
@@ -1,12 +1,9 @@
-tests/cases/conformance/classes/classDeclarations/classAbstractKeyword/classAbstractManyKeywords.ts(1,25): error TS1005: ';' expected.
 tests/cases/conformance/classes/classDeclarations/classAbstractKeyword/classAbstractManyKeywords.ts(3,1): error TS1128: Declaration or statement expected.
 tests/cases/conformance/classes/classDeclarations/classAbstractKeyword/classAbstractManyKeywords.ts(4,17): error TS1005: '=' expected.
 
 
-==== tests/cases/conformance/classes/classDeclarations/classAbstractKeyword/classAbstractManyKeywords.ts (3 errors) ====
+==== tests/cases/conformance/classes/classDeclarations/classAbstractKeyword/classAbstractManyKeywords.ts (2 errors) ====
```
Comment (DanielRosenwasser) [inline tests/baselines/reference/classAbstractManyKeywords.errors.txt]: Wow nice test name @aozgaa

**Comment by DanielRosenwasser** [comment]
👍 LGTM

**Comment by aluanhaddad** [comment]
Awesome! 

It is a shame this doesn't support interfaces. I tend to abide by the rule that says if you have only one export, it should be the default, and I apply this even to interfaces.

Regardless, I'm really glad to see this. Thank you!

**Comment by erikmcc** [comment]
I'm planning on fixing that next. I just wanted to see what the PR
submission process was like before investing too much time.

**Comment by BehindTheMath** [comment]
@erikmcc Does it make sense that I'm still getting an error if I use the `export =` syntax (for example, `export = abstract class A`)?

**Comment by mhegazy** [comment]
We are not planning on changing the `export = ` at the time being.

**Pull request merged**